File: Leetcode/1768. Merge Strings Alternately/solution.rb
def merge_alternately(word1, word2)
  mergedStr = ""

  sizeWordOne = word1.size
  sizeWordTwo = word2.size

  majorLength = sizeWordOne > sizeWordTwo ? sizeWordOne : sizeWordTwo

  count = 0

  while count < majorLength

    if word1[count] && word2[count]
      mergedStr += word1[count] + word2[count]
      count += 1
    elsif word1[count] && !word2[count]
      mergedStr += word1[count]
      count += 1
    else
      mergedStr += word2[count]
      count += 1
    end

  end

  mergedStr
end

puts merge_alternately("abc", "pqr") # apbqr
puts merge_alternately("ab", "pqrs") # apbqrs

File: Leetcode/1768. Merge Strings Alternately/solution.js
const mergeAlternately = (word1, word2) => {
  let mergedStr = ""

  const word1Size = word1.length
  const word2Size = word2.length

  const majorSize = word1Size > word2Size ? word1Size : word2Size

  count = 0

  while(count < majorSize) {
    if(word1[count] && word2[count]) {
      mergedStr += word1[count] + word2[count]
      count ++
    } else if (word1[count] && !word2[count]) {
      mergedStr += word1[count]
      count ++
    } else {
      mergedStr += word2[count]
      count++
    }
  }

  return mergedStr
}


console.log(mergeAlternately("abc", "pqr"))
